src/server/project.ts

```diff
@@ -1841,8 +1857,10 @@ namespace ts.server {
                     moduleResolutionHost));
 
                 for (const resolution of resolutions) {
-                    if (resolution.resolvedTypeReferenceDirective?.resolvedFileName && !hostProject.getCurrentProgram()!.getSourceFile(resolution.resolvedTypeReferenceDirective.resolvedFileName)) {
-                        rootNames = append(rootNames, resolution.resolvedTypeReferenceDirective.resolvedFileName);
+                    if (!resolution.resolvedTypeReferenceDirective?.resolvedFileName) continue;
+                    const { resolvedFileName } = resolution.resolvedTypeReferenceDirective;
+                    if (!hostProject.getCurrentProgram()!.getSourceFile(resolvedFileName)) {
+                        rootNames = append(rootNames, moduleResolutionHost.realpath?.(resolvedFileName) || resolvedFileName);

```

Comment:
```suggestion
                    const { resolvedFileName } = resolution.resolvedTypeReferenceDirective;
                    const fileName = moduleResolutionHost.realpath?.(resolvedFileName) || resolvedFileName; 
                    if (!hostProject.getCurrentProgram()!.getSourceFile(fileName)) {
                        rootNames = append(rootNames, fileName);
```

Reply:
Do you not need to check sourceFile by symlink as well as realpath?

Reply:
Good catch 👍